GH-672: review 4

bertllll opened the pull request.

masqrauder committed:
Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

masqrauder committed:
Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

masqrauder committed:
GH-600: Handle expected batch, got single response

masqrauder committed:
GH-600: Handle expected batch, got single response

utkarshg6 committed:
GH-701: remove deprecated (September 2021) manifest keys from Cargo.toml files (#301)

bertllll committed:
GH-719: Replace BlockchainInterfaceClandestine with BlockchainInterfaceNull (#307)

* GH-719: Blockchain interfaces slightly modified, mainly naming. The clandestine was replaced with BlockchainInterfaceNull

* GH-719: review issues fixed

* GH-719: review two answered

* GH-719: review 3 answered

* GH-719: additional changes for review 3

* GH-719: web3 ABI reformated

---------

Co-authored-by: Bert <Bert@Bert.com>

masqrauder committed:
Resolve rebase issues

masqrauder committed:
Resolve rebase issues

utkarshg6 committed:
GH-691: Integrate `--min-hops` with `PersistentConfiguration` (#294)

* GH-691: database migration for 7 to 8

* GH-691: update CURRENT_SCHEMA_VERSION to 8

* GH-691: add test to set the min_hops_count inside PersistentConfiguration

* GH-691: fix tests in actor_system_factory.rs

* GH-691: add the ability to retrieve value of min_hops_count from the database

* GH-691: add tests for validating min hops count

* GH-691: test drive code for validating the min_hops_count value from the persistent configuration

* GH-691: minor code improvements

* GH-691: remove second declaration of CURRENT_SCHEMA_VERSION

* GH-691: remove unnecessary comments

* GH-691: improve validate_database_min_hops_count()

* GH-691: fixed test cleanup_after_deceased_clients_integration

* GH-691: add todos to where we should set the min_hops_count

* GH-691: set min_hops_count value inside configure_database()

* GH-691: remove code for setting min_hops_count inside actor_system_factory.rs

* GH-691: set min_hops_count default value while initializing database

* GH-691: change how min_hops_count is handled inside make_neighborhood_config()

* GH-691: instead of defaulting min_hops_count inside clap, default it in db

* GH-691: feed min_hops_count value inside ConfigDaoNull

* GH-691: add test for checking whether the node panics when the min_hops_count value is not found

* GH-691: remove clippy warnings

* GH-691: remove redundant code

* GH-691: more improvements to  validate_database_min_hops_count()

* GH-691: Review 1 (#300)

* GH-691: minor changes in persistent_configuration.rs

* GH-691: Display for Hops can be simplified

* feat: panic when value of min_hops_count is not present in db

* GH-591: remove setting a result for set_min_hops_count inside actor_system_factory.rs

* GH-691: improvements in migration_7_to_8.rs

* fix: the todo has been changed to a comment after discussion in the dev team

* GH-691: Review 2 (#302)

* GH-691: rename min_hops_count to min_hops everywhere

* GH-691: implement the computed_default()

* GH-691: implement the computed_default() for min hops

* GH-691: replace min hops value to the the one in db

* GH-691: rename DEFAULT_MIN_HOPS_COUNT and MIN_HOPS_COUNT_FOR_TEST to DEFAULT_MIN_HOPS and MIN_HOPS_FOR_TEST respectively

* GH-691: remove unnecessary test

* GH-691: Review 3 (#304)

* GH-691: convert panic into error log inside the setup_reporter.rs

* GH-691: use Default instead of Configured if the default min hops value is found in db

* GH-691: change debug to info in the validation

* GH-691: use AssertionsMessage for asserting when the validation does not need to replace the value

* GH-691: remove use of min_hops_params from the tests in Neighborhood

* GH-691: remove min_hops_params

* GH-691: fix typo in test name

* GH-691: improve the log statement

* GH-691: fix the remaining renames (#312)

utkarshg6 committed:
GH-727: Don't run actions when PR is in Draft Mode (#319)

masqrauder committed:
GH-600: Adapt to new Pokt message changes

masqrauder committed:
GH-600: Adapt to new Pokt message changes

masqrauder committed:
Use start_block in the event of an error

masqrauder committed:
Use start_block in the event of an error

Commit:
GH-672: first batch of fixed items from the review 3

Commit:
GH-672: another batch of fixes

Commit:
GH-672: refactoring some complicated test concepts for actor_system_factory and adding one test for blockchain_bridge of that kind

Commit:
GH-672: more refactoring and little changes

Commit:
GH-672: fixing stuff araound blockchain_interface web3 and its initialization

Commit:
GH-672: savepoint

Commit:
GH-672: renaming

Commit:
GH-672: almost all things done; AgentDigest fully in

Commit:
GH-672: finished last few missing places

Commit:
GH-672: reworking comments and found a little mistake

Commit:
GH-672: match stop condition turned into easier one to use

Commit:
GH-672: protected payables implemented

Commit:
GH-672: until now the protected payables were implemented badly; improved design

Commit:
GH-672: compiling, but many failing tests

Commit:
GH-672: still fixing tests, but a lot of them passing now

masqrauder committed:
GH-600: Return blockchain scan error to update max_block_count

masqrauder committed:
GH-600: Return blockchain scan error to update max_block_count

Commit:
GH-672: halting hard with thoughts about implementation of the build_blockcahin_agent_method

Commit:
GH-672: blockchain interface onward and backwards move; messy rearrangement; wanna make a savepoint before trying cargo fix

Commit:
GH-672: all tests good; saving before cargo fix

bertllll committed:
GH-580: test for data_directory_from_context with opt dir and implementation … (#248)

* test for data_directory_from_context with opt dir and implementation in fn

* fixed naming and optimizing call for add_chain_specific_directories()

* add_chain_specific_directories to tests in  bootstrapper.rs, mod.rs, node_configuraton_standard.rs

* calculate_configured_setup returns None -> panicked at 'called  on a  value'

* formatting

* cleaning misstake

* commit changes - move to laptop

* fix tests failing on the chain specific data direcotry

* changes based on Berts review comments

* formatting

* remove todo causing crash

* half way to fix cleanup_after_deceased_clients_integration

* failing on creating log file in node_logs_panic_integration

* change joins for add_chain_specific_directories fn

* remove unused import

* add_chain_specific_directories moved to masq_lib::utils, introduced new function node_chain_specific_directory for multinode tests

* remove comments

* remove comments, replace joins wtih add_chain_specific_directories fn, test for chain specific directories uncompleted but passing

* Revert "remove unused import"

This reverts commit c0f205a2cd6c5ecb1d53e48cf2763ffacd1455f4.

* pull and resolve conflicts

* vec! with foreach for test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* add more scenarios for test with custom data_directory_opt in test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* continue of solving Bert's comments, start of modifying UI to show full path for user specified data-directory

* try to retrieve chain-name from inner.values

* chain_name collecting with map().collect, value_value collecting with match

* move match for value_value to separate fn to clar dump_setup

* chainge test for right fn calls, added message about changed data-directory to match chain parameter

* change of chain_specific_data_directory to add alway chain_specific_dir to path set by user, so chain dir could be duplicated, finalize test for masq UI to check various UiSetupResponsValueStatus

* new integration test for specific data_directory ensure_data_directory_has_specific_chain_direcotry_within

* fix Default arm for directory path enrichment with chain specific directory

* integration test to ensure setup command do the right job to add chain specific directory

* change data-directory, comparsion and panic messages in chain_specific_directory_test_integration test, remove comments inn setup_command

* fix formatting for expected in chain_specific_directory test, remove unused import from setup_command

* finalize test for setup_reporter calculate_fundamentals_with_chain_specific_dir

* commented code to test in linux

* compilable version to test in linux

* move changes to desktop

* for bert from my linux machine

* GH-580: showing the right direction

* fixing more tests

* GH-580: ongoing repairment in setup_reporter

* GH-580: fixed data directory niches that were causing a lot of troubles

* remove unvanted println

* formatting

* fixed test calculate_setup_with_chain_specific_dir to pass on mac, resolved last Berts comments

* remove unused imports and chain from mock node

* remove unused code from setup_command

* resolving review comments - add closure for retriving data-dir and chain

* resolving comments: cleaning setup_command tests, cleaning setup_reporter:get_modified_setup, fix misspe<lls

* try to send less data to determine_setup_value_of_set_data_directory fn

* remove unused data from SetupReporterReal::determine_setup_value_of_set_data_directory

* formatting

* fixed test for get_modified_setup::get_modified_setup_user_specified_data_directory_depends_on_new_chain_on_success

* resolved review comments on ensure_data_directory_has_specific_chain_directory_within_integration test

* split of test calculate_setup_with_chain_specific_dir into two separate tests

* resolving review comments, renaming, optimization, fixing tests

* resolving review comments - mainly renaming

* Half way to get rid of DirsWrapper from go fn in DumpConfigRunnerReal

* remove DirsWrapper from arguments to DumpConfigRunner for DumpConfigRunnerReal

* fixed windows tests for determine_fundamentals

* renaming functions, removing TODOs and removing unnecesary attributes from test

* fixed node_chain_specific_data_directory and impl MASQRealNode plus try to fix node_creates_log_file_with_heading_integration - continue next

* fixed node_creates_log_file_with_heading_integration

* remove unused import and debug print, tests failing in ci/all.sh but passing when run from cmd line

* fix paths in tests for windows: setup_reporter

* fix windows paths for test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* fix setup_command and broadcast_handler tests: added missing data-directory and chain parameters to config

* setup_command removign ropsten and correcting UiResponseValue, broadcast_handler: removing unwanted piece of code, shared_schema: DATA_DIRECTORY_HELP, setup_reporter: fix test

* correction of DATA_DIRECTORY_HELP

* lazy_static for STANDARD_ARG_LIST in shared_schema

* separate help for --data-directory in Node, Config Dump and UI

* remove duplication of data-directory clap app argument, add dirs to masq_lib to compute static constant DATA_DIRECTORY_DAEMON_HELP, fix broadcast_hadnler test

* move data-directory back to shared schema::shared_app::head

* remove typo misstake

* apply Dans comments on review 6, put code in the order, formatting and moving code thru directories and files

* remove unused import

* resolve DATA_DIRECTORY_DAEMON_HELP for masq UI

---------

Co-authored-by: Vojtěch Parkán <vojtechparkan@Vojtechs-iMac.home>
Co-authored-by: Vojtěch Parkán <vojtechparkan@Vojtechs-iMac.local>
Co-authored-by: Vojtěch Parkán <czarte@gmail.com>
Co-authored-by: Bert <Bert@Bert.com>

Commit:
GH-672: roughly ready code; let's review now

Commit:
GH-672: renaming in accountant folders

Commit:
GH-672: reverting some renaming in favor of another branch taking care of this

Commit:
GH-672: interim commit

Commit:
GH-672: little fixes and adding more in to some null methods with todos

Commit:
GH-672: polishing; savepoint before moving get_transaction_receipt to plain_rpc

Commit:
GH-672: improved details

bertllll committed:
Little renamings (#333)

* little renaiming

* little renaming: formatting

---------

Co-authored-by: Bert <Bert@Bert.com>

dnwiebe commented on multinode_integration_tests/tests/verify_bill_payment.rs:
```diff
@@ -61,13 +61,8 @@ fn verify_bill_payment() {
-        web3_gas_const_part,
-    );
+    let blockchain_interface =
+        BlockchainInterfaceNonClandestine::new(http, _event_loop_handle, cluster.chain);
```
`_event_loop_handle` doesn't need an underscore prefix

dnwiebe commented on node/src/accountant/database_access_objects/receivable_dao.rs:
```diff
@@ -419,6 +419,7 @@ mod tests {
         assert_account_creation_fn_fails_on_finding_wrong_columns_and_value_types,
         make_receivable_account,
     };
+    use crate::blockchain::blockchain_interface::BlockchainTransaction;
```
Huh: why didn't Clippy complain about this?

dnwiebe commented on node/src/accountant/payment_adjuster.rs:
```diff
@@ -73,63 +73,35 @@ pub enum AnalysisError {}
-            agent: Box::new(agent_for_enough),
+        let mut payable = make_payable_account(111);
+        payable.balance_wei = 100_000_000;
+        let agent = BlockahinAgentMock::default();
```
Misspelled

dnwiebe commented on node/src/accountant/scanners/mid_scan_msg_handling/payable_scanner/agent_null.rs:
```diff
@@ -0,0 +1,193 @@
+        U256::zero()
+    }
+
+    test_only!(fn dup(&self) -> Box<dyn BlockchainAgent> {
```
This use of `test_only!` doesn't seem clearer than a use of `#[cfg(test)]`.

dnwiebe commented on node/src/accountant/scanners/mid_scan_msg_handling/payable_scanner/agent_web3.rs:
```diff
@@ -0,0 +1,140 @@
+
+// 64 * (64 - 12) ... std transaction has data of 64 bytes and 12 bytes are never used with us;
+// each non-zero byte costs 64 units of gas
+pub const WEB3_MAXIMAL_GAS_LIMIT_MARGIN: u64 = 3328;
```
Does this constant still need to be `pub`?

dnwiebe commented on node/src/accountant/scanners/mid_scan_msg_handling/payable_scanner/blockchain_agent.rs:
```diff
@@ -0,0 +1,44 @@
+    fn consuming_wallet(&self) -> &Wallet;
+    fn pending_transaction_id(&self) -> U256;
+
+    test_only!(
```
You said I should put in a comment saying that you should spend more time thinking about this, whatever that means.

dnwiebe commented on node/src/accountant/scanners/mid_scan_msg_handling/payable_scanner/mod.rs:
```diff
@@ -0,0 +1,71 @@
+{
+}
+
+pub trait MidScanPayableHandlingScanner {
```
__EFFORTS FOR A BETTER NAME__

Things we're doing here:
Checking solvency
Possibly seeking assistance or advice
Constructing outbound payments

Things we're not doing here:
Scanning
Paying
Adjusting

`SolvencySensitivePaymentInstructor`

dnwiebe commented on node/src/accountant/scanners/mid_scan_msg_handling/payable_scanner/setup_msg.rs:
```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
```
Maybe a better name for this file?

dnwiebe commented on node/src/accountant/scanners/mod.rs:
```diff
@@ -251,43 +248,44 @@ impl Scanner<PayablePaymentsSetupMsgPayload, SentPayables> for PayableScanner {
-                    agent: msg.agent,
-                    response_skeleton_opt: msg.payload.response_skeleton_opt,
-                }))
+                let protected = msg.qualified_payables;
```
Maybe this field should be called `protected_payables` instead.

dnwiebe commented on node/src/accountant/scanners/scanners_utils.rs:
```diff
@@ -550,7 +546,7 @@ mod tests {
             hash: make_tx_hash(123),
         };
-        let bad_rpc_call = RpcPayableFailure {
+        let bad_rpc_call = RpcFailurePayables {
```
How about `RpcPayablesFailure`? Also, in case it matters, the English idiom here would probably be, "That jackass screwed it up."

dnwiebe commented on node/src/accountant/test_utils.rs:
```diff
@@ -1398,87 +1399,72 @@ impl PayableThresholdsGaugeMock {
     }
 }
 
+pub fn protect_payables_in_test(payables: Vec<PayableAccount>) -> ProtectedPayables {
```
I don't like this duplicated code, but I agree that an alternative is hard to find.

dnwiebe commented on node/src/accountant/test_utils.rs:
```diff
@@ -1680,124 +1672,51 @@ impl ScanSchedulers {
-        self.conclude_required_fee_per_computed_unit_params = params.clone();
-        self
+    fn consuming_wallet(&self) -> &Wallet {
+        todo!()
```
Puzzling that this wasn't used in a test yet.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
```diff
@@ -0,0 +1,294 @@
+        let port = find_free_port();
+        let _test_server = TestServer::start(
+            port,
+            vec![br#"{"jsonrpc":"2.0","id":0,"result":"0xFFFF"}"#.to_vec()],
```
Could you  use a more distinctive balance here? 0xFFFF is common enough that it could theoretically be some kind of sentinel value. 0xDEADBEEF, for example, probably isn't.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
```diff
@@ -0,0 +1,294 @@
+        let subject = make_subject(transport, chain);
+
+        let result = subject.get_transaction_fee_balance(&Wallet::new(
+            "0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fQ",
```
Can you make this wallet address a little more obviously invalid? If you don't want a wallet address like `i am obviously invalid`, then you could just put in a comment drawing the reader's attention to that `Q` at the end.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
```diff
@@ -0,0 +1,294 @@
+    fn get_masq_balance_works() {
+        let port = find_free_port();
+        let _test_server = TestServer::start (port, vec![
+            br#"{"jsonrpc":"2.0","id":0,"result":"0x000000000000000000000000000000000000000000000000000000000000FFFF"}"#.to_vec()
```
Again, a more distinctive value would be more reassuring.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
```diff
@@ -0,0 +1,294 @@
+            )
+            .unwrap();
+
+        assert_eq!(result, U256::from(65_535));
```
I'd like to see an assertion about what the server received.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
```diff
@@ -0,0 +1,294 @@
+            )
+            .unwrap();
+
+        assert_eq!(result, U256::from(65_535));
```
I'd like to see an assertion about what the server received.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
```diff
@@ -0,0 +1,294 @@
+        let subject = make_subject(transport, chain);
+
+        let result =
+            subject.get_masq_balance(&Wallet::new("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fQ"));
```
Again, a more obviously-invalid wallet would be helpful.

dnwiebe commented on node/src/blockchain/blockchain_interface/mod.rs:
```diff
@@ -0,0 +1,265 @@
+    fn helpers(&self) -> &dyn RPCHelpers;
+}
+
+#[derive(Clone, Debug, Eq, Message, PartialEq)]
```
Does this need to be an actor `Message`?

dnwiebe commented on node/src/blockchain/blockchain_interface/mod.rs:
```diff
@@ -0,0 +1,265 @@
+            }
+            Self::TransactionCount(blockchain_err) => write!(
+                f,
+                "Transaction count fetching failed for: {}",
```
How does that "for" read in the error message? I don't know what sort of messages you get in `blockchain_err`, but it seems as though it might be a little clumsy. It's a small nit to pick, but it might be better without the "for," or perhaps with something like "because of" or "due to" instead.

dnwiebe commented on node/src/blockchain/test_utils.rs:
```diff
@@ -101,65 +98,47 @@ impl BlockchainInterface for BlockchainInterfaceMock {
 
 impl BlockchainInterfaceMock {
-    //seems like never-used
+    // TODO seems like never-used
```
This is pretty simple. If it isn't used, you can probably delete it with confidence that someone who needs it in the future can write it again.

dnwiebe commented on node/src/sub_lib/blockchain_bridge.rs:
```diff
@@ -18,14 +18,16 @@ use web3::types::U256;
     pub bind: Recipient<BindMessage>,
     pub outbound_payments_instructions: Recipient<OutboundPaymentsInstructions>,
-    pub initial_payable_payment_setup_msg: Recipient<PayablePaymentsSetupMsgPayload>,
+    pub qualified_paybles_message: Recipient<QualifiedPayablesMessage>,
```
Misspelled.

dnwiebe commented on node/src/sub_lib/blockchain_bridge.rs:
```diff
@@ -18,14 +18,16 @@ use web3::types::U256;
 pub struct BlockchainBridgeConfig {
     pub blockchain_service_url_opt: Option<String>,
     pub chain: Chain,
+    //TODO: actually totally ignored during the setup of the BlockchainBridge actor!
```
Did you see the movie _Caddyshack?_

"Well? We're waiting!"

dnwiebe reviewed (commented):
Resume at node/src/sub_lib/blockchain_bridge.rs:49 green

dnwiebe commented on node/src/sub_lib/blockchain_bridge.rs:
```diff
@@ -80,12 +100,20 @@ mod tests {
-        assert_eq!(web3_gas_limit_const_part(Chain::EthRopsten), 55_000);
-        assert_eq!(web3_gas_limit_const_part(Chain::Dev), 55_000)
+    fn blockchain_bridge_subs_factory_produces_proper_subs() {
+        let subject = BlockchainBridgeSubsFactory {};
```
I was confused about why this was necessary, instead of just using `BlockchainBridge::make_subs_from()` directory, but I was assured I would understand when I looked at `actor_system_factory.rs`. Maybe I will. If I do, ignore this.

dnwiebe commented on node/src/test_utils/mod.rs:
```diff
@@ -1047,15 +1046,19 @@ pub mod unshared_test_utils {
+                        // capture and examine these args in assertions is to receive the ArbitraryIdStamp of the given
+                        // argument.
+                        // If such strategy is once decided for, transfers of this id will have to happen in all the tests
+                        // relying on this mock, while also calling the meant method. So even in cases where we certainly
```
How about "intended method?" Means the same, but sounds a little smoother.

Also--I'm making an assumption here about what you mean, which may not be correct--try putting a comma instead of a period after "checking that id," and a comma after "we ignored that."

dnwiebe commented on node/src/test_utils/recorder_stop_conditions.rs:
```diff
@@ -117,17 +120,37 @@ impl StopCondition {
+impl<T: Send> PartialEq for PretendedMatchableWrapper<T> {
+    fn eq(&self, _other: &Self) -> bool {
+        panic!(
+            r#"You requested a stop condition of the "precise match" for
```
Can you use `StopCondition::StopOnMatch` here instead of "precise match?"

dnwiebe commented on node/src/actor_system_factory.rs:
```diff
@@ -157,13 +150,14 @@ impl ActorSystemFactoryTools for ActorSystemFactoryToolsReal {
             &db_initializer,
             &BannedCacheLoaderReal {},
-            &AccountantSubsFactoryReal {},
+            &AccountantSubsFactory {},
```
Should this go back to `AccountantSubsFactoryReal`?

dnwiebe commented on node/src/actor_system_factory.rs:
```diff
@@ -888,7 +901,7 @@ mod tests {
+            _accountant_subs_factory: &dyn SubsFactory<Accountant, AccountantSubs>,
         ) -> AccountantSubs {
             self.parameters
                 .accountant_params
```
This mark is in the wrong place. You noticed that this method ends with `make_accountant_subs_from_recorder()`, while other similar methods end with `make_xxx_subs_from()`, and you wondered whether those other methods should be renamed to `make_xxx_subs_from_recorder()`.

dnwiebe commented on node/src/actor_system_factory.rs:
```diff
@@ -1936,169 +2087,134 @@ mod tests {
-    fn our_big_int_sqlite_functions_are_linked_to_receivable_dao_within_accountant() {
-        //condition: .new_delinquencies() still encompasses our user defined functions (that's why a formal check opens this test)
+    fn accountant_has_db_connection_that_can_make_use_of_our_own_sqlite_functions() {
+        //precondition: .new_delinquencies() still encompasses our meant functions, otherwise the test is false-positive
```
Intended?

dnwiebe commented on node/src/actor_system_factory.rs:
```diff
@@ -1936,169 +2087,134 @@ mod tests {
 
         let accountant_addr = addr_rv.try_recv().unwrap();
-        //this message also stops the system after the check
+        //the message stops also the system
```
"Also stops" is better than "stops also"

dnwiebe commented on node/src/actor_system_factory.rs:
```diff
@@ -1936,169 +2087,134 @@ mod tests {
         );
+
+        let blockchain_addr = addr_rv.try_recv().unwrap();
+        //the message stops also the system
```
"Also stops" is better than "stops also"

dnwiebe commented on node/src/actor_system_factory.rs:
```diff
@@ -1936,169 +2087,134 @@ mod tests {
-            alias_cryptde_public_key_after,
-            alias_cryptde_public_key_before
+    #[test]
+    fn blockchain_bridge_connectivity_check() {
```
You said you thought maybe you could make this test neater.

bertllll commented on node/src/accountant/scanners/mod.rs:
```diff
@@ -499,6 +497,19 @@ impl PayableScanner {
+        unsafe {
+            transmute::<Vec<u8>, Vec<PayableAccount>>(protected.0)
+        }
+    }
```
Add in another function pair, which will make up the inside of these two. They, for change, will be static function but with some kind of restricted visibility. Probably `pub (in crate::accountant)`. Then go to the right `test_utils` location and use it again in the test only method that is supposed to act a as helper in tests where we need to construct the protected payables but we shouldn't be able to in the production version.

bertllll commented on node/src/actor_system_factory.rs:
```diff
@@ -1936,169 +2087,134 @@ mod tests {
+        let blockchain_addr = addr_rv.try_recv().unwrap();
+        //the message stops also the system
+        blockchain_addr
+            .try_send(assertion_msg_to_test_blockchain_bridge_connectivity(
```
Think about using a public manifest constant instead of a function call. It allows to initiate a struct in its literal form which is how we usually construct the message.

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -547,11 +552,8 @@ impl Accountant {
-                .blockchain_bridge
-                .initial_payable_payment_setup_msg,
-        );
+        self.payable_payments_setup_msg_payload_sub_opt =
```
Field name could change.

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -1430,42 +1401,45 @@ mod tests {
-                    context_id: 4321,
-                }),
-            },
+        let payable_payments_setup_msg = BlockchainAgentWithContextMessage {
```
Variable name could change.

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -3210,12 +3191,15 @@ mod tests {
+        let persistent_config_id_stamp = ArbitraryIdStamp::new();
+        let persistent_config = PersistentConfigurationMock::default()
+            .set_arbitrary_id_stamp(persistent_config_id_stamp)
+            .gas_price_result(Ok(130));
```
You thought this wasn't necessary.

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -4477,3 +4456,26 @@ mod tests {
+    pub fn assertion_msg_to_test_registration_of_our_sqlite_fns() -> AssertionsMessage<Accountant> {
+        AssertionsMessage {
+            assertions: Box::new(|accountant: &mut Accountant| {
+                // Will crash a test if our user-defined SQLite fns have been unreachable;
```
You thought depending on a panic might not be reliable.

masqrauder committed:
GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

masqrauder committed:
GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

utkarshg6 committed:
GH-745: Macro for creating HashMap (#337)

* macros-experiment: add hashmap!()

* GH-745: gitignore files

* GH-745: untrack file

* GH-745: trigger actions

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
```diff
@@ -213,60 +211,35 @@ impl BlockchainBridge {
-        ))
+        match blockchain_service_url_opt {
+            Some(url) => {
+                let initializer = BlockchainInterfaceInitializer {};
```
It looks to me as though this should be a collaborator that's part of `BlockchainBridge`, and that this function should be a method instead. In that case, we could make `BlockchainInterfaceInitializer` smarter and mock it.

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
```diff
@@ -213,60 +211,35 @@ impl BlockchainBridge {
             bind: recipient!(addr, BindMessage),
             outbound_payments_instructions: recipient!(addr, OutboundPaymentsInstructions),
-            initial_payable_payment_setup_msg: recipient!(addr, PayablePaymentsSetupMsgPayload),
+            qualified_paybles_message: recipient!(addr, QualifiedPayablesMessage),
```
"Paybles" is misspelled. Also, does it sound right to have a recipient whose name ends with "message?" I know the previous version did too, but I think maybe it was wrong as well.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs:
```diff
@@ -350,23 +154,62 @@ where
+    ) -> Result<Box<dyn BlockchainAgent>, String> {
+        macro_rules! err {
+            ($($arg: tt)*) => {
+                Err(format!("Blockchain agent build failed to fetch {}", format!($($arg)*)))
```
You didn't like the word "build" here.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs:
```diff
@@ -907,7 +686,7 @@ mod tests {
         let port = find_free_port();
         #[rustfmt::skip]
-        let test_server = TestServer::start (port, vec![
+            let test_server = TestServer::start (port, vec![
```
This indentation might have been accidental.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs:
```diff
@@ -1768,8 +1465,18 @@ mod tests {
-        assert!(transaction_params.gas <= U256::from(not_above_this_value));
+        assert!(
+            transaction_params.gas >= U256::from(not_under_this_value),
+            "actual gas limit {} isn't above {}",
```
This message is a little misleading; it makes things look like the comparison is > rather than >=. You could say "isn't above or equal to {}", or you could say "is below {}".

dnwiebe reviewed (commented):
Resume at blockchain_interface.rs:1792 red

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs:
```diff
@@ -1789,22 +1496,14 @@ mod tests {
-        let result =
-            subject.send_batch_of_payables(&consuming_wallet, &agent, &recipient, &accounts);
+        let agent = make_initialized_agent(123, consuming_wallet, U256::from(4));
+        let result = subject.send_batch_of_payables(agent, &recipient, &accounts);
```
Blank line before Act...

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs:
```diff
@@ -2552,157 +2011,24 @@ mod tests {
-        TestLogHandler::new()
-            .assert_logs_contain_in_order(vec![expected_log_msg.as_str()].repeat(chains.len()));
+    fn hash_the_smartcontract_transfer_function_signature() {
+        assert_eq!(
```
Reverse this assertion

Commit:
GH-672: mostly done; just a few details left

Commit:
GH-672: abondoning enhancement of blockchain interface initializer...leaving as it is

Commit:
GH-672: unnoticable change for the augly low level tests in actor_system_factory

Commit:
GH-672: blockchain_bridge_shared_test_simplified

Commit:
GH-672: ready for review 5

masqrauder committed:
GH-600: Add test for scan time out

masqrauder committed:
GH-600: Add test for scan time out

masqrauder committed:
GH-600: Revert scanner changes

masqrauder committed:
GH-600: Revert scanner changes

utkarshg6 committed:
GH-692:  Allow UI to configure --min-hops during runtime (#316)

* GH-692: ignore file dbnavigator.xml

* GH-692: add tests for command execution of set-configuration

* GH-692: revert the version number

* GH-692: add test for the error case

* GH-692: improve tests to work on macos

* GH-692: improve tests to work on macos

* GH-692: ignore dbnavigator.xml

* GH-692: remove the redundant gitignore line

* GH-692: add tests for setting min_hops using set-configuration

* GH-692: add TODOs

* GH-692: migrate the Consuming Wallet change and Password Change to a new SetConfigurationMessage

* GH-692: update field names inside SetConfigurationMessage and variants name inside ConfigurationChange

* GH-692: update field names inside SetConfigurationMessage and variants name inside ConfigurationChange

* GH-692: write test can_update_min_hops_with_set_configuration_msg

* GH-692: set min hops value in db inside node_configurator/configurator.rs

* GH-692: only change value of min hops in neighborhood using SetConfigurationMessage

* GH-691: write more tests for setting min hops inside the node_configurator/configurator.rs

* GH-692: an attempt to remove the SetConsumingWalletMessage

* GH-692: remove SetConsumingWalletMessage from all the other places because it's a dead code

* GH-692: remove the SetConsumingWalletMessage completely

* GH-692: remove NewPasswordMessage

* GH-692: replace NewPasswordMessage with SetConfigurationMessage

* GH-692: add test for UpdateNewPassword inside configurator.rs

* GH-692: write more tests for the configurator.rs

* GH-692: improve test handle_set_configuration_handles_failure_on_min_hops_database_issue

* GH-692: change name of message to ConfigurationChangeMessage

* GH-691: implement calculate_db_patch_size()

* GH-692: add the fn to set_min_hops_and_db_patch_size()

* GH-692: set db_patch_size while setting min_hops

* GH-692: completely remove NewPasswordMessage

* GH-692: drive code for setting up patch size inside NeighborhoodMetadata

* GH-692: use db_patch_size or patch_size instead of min_hops while computing patch

* GH-692: remove redundant functions

* GH-692: formatting changes

* GH-692: ignore the stray test - ci/all.sh is passing

* GH-692: migrate min_hops test to a new file in multinode tests

* GH-692: attempt to write multinode test

* GH-692: why is this working? and not the previous version

* GH-692: improve multinode test

* GH-692: test drive the RouteQuery after the min hops has been changed

* GH-692: refactor the neighborhood/mod.rs file

* GH-692: min hops multinode test is passing

* GH-692: improve multinode test with better comments and refactoring

* GH-692: improve logging inside Neighborhood and Configurator

* GH-692: another formatting changes

* GH-692: allow overall connection stage to regress

* GH-692: regress stage of OverallConnectionStatus and send a msg to the UI

* GH-692: assert that the ui_gateway received a message for regression

* GH-692
: remove unnecessary warnings

* GH-692: test hardening

* GH-692: reorder msg inside recorder.rs

* GH-692: minor improvements in multinode tests

* GH-692: rename file

* GH-692: few fn renames

* GH-692: trigger actions

* GH-692: remove unused import from multinode test and increase timeout

* GH-692: regress ocs stage only if the initial stage is RouteFound

* GH-692: Review 1 (#334)

* GH-692: add the public constant DEFAULT_TRANSACT_TIMEOUT_MILLIS for the production code

* GH-692: add another constant for the tests

* GH-692: reduce total number of nodes for the min hops data routing test

* GH-692: change the comment for the multinode test

* GH-692: replace the constant for db_patch_size and use the different name everywhere

* GH-692: put code back for NewPasswordMessage

* GH-692: minor changes

* GH-692: add error logging when the UiSetConfigurationRequest causes an error

* GH-692: test fix and remove todo

* GH-692: fix test set_configuration_command_throws_err_for_invalid_arg

* GH-692: remove UpdateNewPassword variant

* GH-692: add better error msg in case the assert statement fails

* GH-692: fix using --gas-price and --min-hops during runtime without any value

* GH-692: required is actually not required

* GH-692: set_configurationify() changes min_values to 1 instead of 0

* GH-692: formatting

Commit:
GH-672: interim commit

Commit:
GH-672: finished review 5

Commit:
GH-672: merge with master

Commit:
GH-672: review 6 answered

Commit:
GH-672: taking over code from master + small fixes

Commit:
GH-672: fixes in multinode tests

Commit:
GH-672: little cosmetics

bertllll committed:
GH-751: Hot fix for wrong books keeping at transactions confirmation  (#349)

* GH-751: done

* GH-751: answer to the review

---------

Co-authored-by: Bert <Bert@Bert.com>

Commit:
GH-672: merge with GH-600 completed

Commit:
GH-672: renamings and moves in test code

Commit:
GH-672: rpc halpers into lower level interface

masqrauder committed:
GH-600: Address review feedback

Commit:
GH-672: incorporated new changes from GH-600

Commit:
GH-672: ending a big rearrangement; stable with all the tests again

Commit:
GH-672: more refactoring

Commit:
GH-672: missing import

Commit:
GH-672: review 7 answered

masqrauder committed:
GH-600: Add get_block_number (#193)

* Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

* GH-600: Handle expected batch, got single response

* Resolve rebase issues

* GH-600: Adapt to new Pokt message changes

* Use start_block in the event of an error

* GH-600: Return blockchain scan error to update max_block_count

* GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

* GH-600: Add test for scan time out

* GH-600: Revert scanner changes

* GH-600: Address review feedback

* GH-600: Address review 6 feedback

* GH-600: Use From traits to be concise

Commit:
GH-672: merged master in

Commit:
GH-672: removed a not suitable comment

Commit:
GH-672: review 8 answered

Commit:
GH-672: lower interface renames
